src/solvers/bitwuzla/bitwuzla_conv.cpp

```diff
@@ -0,0 +1,920 @@
+    boolean_sort);
+}
+
+smt_astt bitwuzla_convt::mk_neq(smt_astt a, smt_astt b)
+{
+  assert(a->sort->get_data_width() == b->sort->get_data_width());
+  return new_ast(
+    bitwuzla_mk_term2(
+      bitw,
+      BITWUZLA_KIND_DISTINCT,
+      to_solver_smt_ast<bitw_smt_ast>(a)->a,
+      to_solver_smt_ast<bitw_smt_ast>(b)->a),
+    boolean_sort);
+}
+
+smt_astt bitwuzla_convt::mk_store(smt_astt a, smt_astt b, smt_astt c)
+{
+  assert(a->sort->id == SMT_SORT_ARRAY);
+  assert(a->sort->get_domain_width() == b->sort->get_data_width());
+  assert(
+    a->sort->get_range_sort()->get_data_width() == c->sort->get_data_width());
+  return new_ast(
+    bitwuzla_mk_term3(
+      bitw,
+      BITWUZLA_KIND_ARRAY_STORE,
+      to_solver_smt_ast<bitw_smt_ast>(a)->a,
+      to_solver_smt_ast<bitw_smt_ast>(b)->a,
+      to_solver_smt_ast<bitw_smt_ast>(c)->a),
+    a->sort);
+}
+
+smt_astt bitwuzla_convt::mk_select(smt_astt a, smt_astt b)
+{
+  assert(a->sort->id == SMT_SORT_ARRAY);
+  assert(a->sort->get_domain_width() == b->sort->get_data_width());
+  return new_ast(
+    bitwuzla_mk_term2(
+      bitw,
+      BITWUZLA_KIND_ARRAY_SELECT,
+      to_solver_smt_ast<bitw_smt_ast>(a)->a,
+      to_solver_smt_ast<bitw_smt_ast>(b)->a),
+    a->sort->get_range_sort());
+}
+
+smt_astt bitwuzla_convt::mk_smt_int(const BigInt &theint [[gnu::unused]])
+{
+  std::cerr << "Bitwuzla can't create integer sorts" << std::endl;
+  abort();
+}
+
+smt_astt bitwuzla_convt::mk_smt_real(const std::string &str [[gnu::unused]])
+{
+  std::cerr << "Bitwuzla can't create Real sorts" << std::endl;
+  abort();
+}
+
+smt_astt bitwuzla_convt::mk_smt_bv(const BigInt &theint, smt_sortt s)
+{
+  // Fedor: double check the second argument of the function
+  BitwuzlaSort *bitwsort = bitwuzla_mk_bv_sort(bitw, s->get_data_width());
+  return new_ast(
+    bitwuzla_mk_bv_value(
+      bitw,
+      bitwsort,
+      integer2binary(theint, s->get_data_width()).c_str(),
+      BITWUZLA_BV_BASE_BIN),
+    s);
+}
+
+smt_astt bitwuzla_convt::mk_smt_bool(bool val)
+{
+  BitwuzlaTerm *node = (val) ? bitwuzla_mk_true(bitw) : bitwuzla_mk_false(bitw);
+  const smt_sort *sort = boolean_sort;
+  return new_ast(node, sort);
+}
+
+smt_astt bitwuzla_convt::mk_array_symbol(
+  const std::string &name,
+  const smt_sort *s,
+  smt_sortt array_subtype [[gnu::unused]])
+{
+  return mk_smt_symbol(name, s);
+}
+
+smt_astt
+bitwuzla_convt::mk_smt_symbol(const std::string &name, const smt_sort *s)
+{
+  symtable_type::iterator it = symtable.find(name);
+  if(it != symtable.end())
+    return it->second;
+
+  BitwuzlaTerm *node;
+
+  switch(s->id)
+  {
+  case SMT_SORT_BV:
+  case SMT_SORT_FIXEDBV:
+  case SMT_SORT_BVFP:
+  case SMT_SORT_BVFP_RM:
+    node = bitwuzla_mk_const(
+      bitw, to_solver_smt_sort<BitwuzlaSort *>(s)->s, name.c_str());
+    break;
+
+  case SMT_SORT_BOOL:
+    node = bitwuzla_mk_const(
+      bitw, to_solver_smt_sort<BitwuzlaSort *>(s)->s, name.c_str());
+    break;
```

Comment:
Looks like there is no diff between bool and the rest, so you can merge the cases.